prometheus/metrics.go

```diff
@@ -20,89 +19,49 @@ var (
-	queryPart := replaceInvalidCharacters(fmt.Sprintf("%s_%s_%s", servicename, namespace, envoyClustername))
 	now := time.Now()
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	var errH, errT error
+	labels := fmt.Sprintf(`app="%s",namespace="%s"`, app, namespace)
 
-	inboundHealthyChan, inboundTotalChan, outboundHealthyChan, outboundTotalChan := make(chan model.SampleValue, len(ports)), make(chan model.SampleValue, len(ports)), make(chan model.SampleValue, len(ports)), make(chan model.SampleValue, len(ports))
-	errChan := make(chan error)
-
-	// Note: metric names below probably depend on some istio configuration.
-	// They should anyway change soon in a more prometheus-friendly way,
-	// see https://github.com/istio/istio/issues/4854 and https://github.com/istio/istio/pull/5069
-
-	for _, _port := range ports {
-		// Inbound
-		go func(port int) {
-			vec, err := fetchTimestamp(api, fmt.Sprintf("envoy_cluster_inbound_%d__%s_membership_healthy", port, queryPart), now)
-			if err != nil {
-				errChan <- err
-			} else if len(vec) > 0 {
-				inboundHealthyChan <- vec[0].Value
-			} else {
-				inboundHealthyChan <- 0
-			}
-		}(int(_port))
-		go func(port int) {
-			vec, err := fetchTimestamp(api, fmt.Sprintf("envoy_cluster_inbound_%d__%s_membership_total", port, queryPart), now)
-			if err != nil {
-				errChan <- err
-			} else if len(vec) > 0 {
-				inboundTotalChan <- vec[0].Value
-			} else {
-				inboundTotalChan <- 0
-			}
-		}(int(_port))
-		// Outbound
-		go func(port int) {
-			vec, err := fetchTimestamp(api, fmt.Sprintf("envoy_cluster_outbound_%d__%s_membership_healthy", port, queryPart), now)
-			if err != nil {
-				errChan <- err
-			} else if len(vec) > 0 {
-				outboundHealthyChan <- vec[0].Value
-			} else {
-				outboundHealthyChan <- 0
-			}
-		}(int(_port))
-		go func(port int) {
-			vec, err := fetchTimestamp(api, fmt.Sprintf("envoy_cluster_outbound_%d__%s_membership_total", port, queryPart), now)
-			if err != nil {
-				errChan <- err
-			} else if len(vec) > 0 {
-				outboundTotalChan <- vec[0].Value
-			} else {
-				outboundTotalChan <- 0
-			}
-		}(int(_port))
-	}
+	go func() {
+		defer wg.Done()
+		vec, err := fetchTimestamp(api, "sum(envoy_cluster_membership_healthy{"+labels+"})", now)
+		if err != nil {
+			errH = err
+		} else if len(vec) > 0 {
+			ret.Healthy = int(vec[0].Value)
+		} else {
+			ret.Healthy = 0
+		}
+	}()
 
-	var err error
-	for count := 0; count < len(ports)*4; count++ {
-		select {
-		case v := <-inboundHealthyChan:
-			ret.Inbound.Healthy += int(v)
-		case v := <-inboundTotalChan:
-			ret.Inbound.Total += int(v)
-		case v := <-outboundHealthyChan:
-			ret.Outbound.Healthy += int(v)
-		case v := <-outboundTotalChan:
-			ret.Outbound.Total += int(v)
-		case err = <-errChan:
-			// No op
+	go func() {
+		defer wg.Done()
+		vec, err := fetchTimestamp(api, "sum(envoy_cluster_membership_total{"+labels+"})", now)
```

Comment:
Perhaps this comment https://github.com/kiali/kiali/pull/604#discussion_r243568274 is more appropiate under this line.

Reply:
Not sure if the sum() is correct here.
Looking at the values:
![image](https://user-images.githubusercontent.com/1662329/50347367-9d2eb400-0535-11e9-9176-8be9f8f1e7c7.png)
It sounds like we could use just one single value ?